--- a/cmd/sing-box/cmd_version.go
+++ b/cmd/sing-box/cmd_version.go
@@ -50,6 +50,9 @@ func printVersion(cmd *cobra.Command, ar
 	if tags != "" {
 		version += "Tags: " + tags + "\n"
 	}
+	if revision == "" {
+		revision = C.Revision
+	}
 	if revision != "" {
 		version += "Revision: " + revision + "\n"
 	}
--- a/constant/version.go
+++ b/constant/version.go
@@ -1,3 +1,4 @@
 package constant
 
 var Version = "unknown"
+var Revision = "unknown"
